data/preset/gallery.yaml

```diff
@@ -0,0 +1,555 @@
+                card:
+                  type: Card.Title
+                  properties:
+                    halign: fill
+                    vexpand: true
+            selection:
+              type: Selection.AllSets
+              slots:
+                filter: Filter.Featured
+
+- &selection-home
+  type: Selection.FeaturedFirst
+
+- &selection-set
+  type: Selection.ContentForSet
+
+# Pages
+- &home-page
+  type: Layout.TopMenu
+  slots:
+    top-menu:
+      <<: *main-menu
+    content:
+      shortdef: 'Layout.InfiniteScrolling(expand: true)'
+      references:
+        lazy-load: highlighted
+      slots:
+        content:
+          type: Layout.ResponsiveMargins
+          slots:
+            content:
+              shortdef: 'Layout.Box(orientation: vertical)'
+              slots:
+                contents:
+                  - type: ContentGroup.ContentGroup
+                    slots:
+                      arrangement:
+                        <<: *main-arrangement
+                      selection:
+                        <<: *selection-home
+                        id: CG1
+                        slots:
+                          filter:
+                            type: Filter.Articles
+                  - type: ContentGroup.ContentGroup
+                    slots:
+                      arrangement:
+                        <<: *main-arrangement
+                      selection:
+                        <<: *selection-home
+                        id: CG2
+                        slots:
+                          filter:
+                            type: Filter.Articles
+                            slots:
+                              sub-filter:
+                                type: Filter.OtherGroup
+                                references:
+                                  other: CG1
+                  - type: ContentGroup.ContentGroup
+                    slots:
+                      arrangement:
+                        <<: *main-arrangement
+                      selection:
+                        <<: *selection-home
+                        id: CG3
+                        slots:
+                          filter:
+                            type: Filter.Articles
+                            slots:
+                              sub-filter:
+                                type: Filter.OtherGroup
+                                references:
+                                  other: CG1
+                                slots:
+                                  sub-filter:
+                                    type: Filter.OtherGroup
+                                    references:
+                                      other: CG2
+                  - type: ContentGroup.ContentGroup
+                    slots:
+                      arrangement:
+                        <<: *main-arrangement
+                      selection:
+                        <<: *selection-home
+                        id: CG4
```

Comment:
I think I spotted one issue, AFAIK, the module ids should be unique! So, if you id a module "CG1", every module (regardless of where they are in the app) that references that id would be getting the same one.. but here you are assigning the same id to multiple modules, and that could cause a lot of unexpected behaviors.